relayer/relays/beacon/header/header.go

```diff
@@ -552,12 +552,19 @@ func (h *Header) findLatestCheckPoint(slot uint64) (state.FinalizedHeader, error
 		return beaconState, fmt.Errorf("GetLastFinalizedStateIndex error: %w", err)
 	}
 	startIndex := uint64(lastIndex)
-	endIndex := startIndex + 1
+
+	index := startIndex
 
 	syncCommitteePeriod := h.protocol.Settings.SlotsInEpoch * h.protocol.Settings.EpochsPerSyncCommitteePeriod
 	totalStates := h.protocol.Settings.EpochsPerSyncCommitteePeriod * h.protocol.HeaderRedundancy // Total size of the circular buffer,
 	// https://github.com/paritytech/polkadot-sdk/blob/master/bridges/snowbridge/pallets/ethereum-client/src/lib.rs#L75
-	for index := startIndex; index != endIndex; index = (index - 1 + totalStates) % totalStates {
+	for {
+		index = (index - 1 + totalStates) % totalStates
+		if index == startIndex {
+			log.WithError(err).Debug("unable to find a relevant on-chain header, max iterations reached")
+			break
+		}
+
```

Comment:
Better to move this to the end or we may miss the first item at startIndex.

Reply:
Good catch: https://github.com/Snowfork/snowbridge/pull/1304/commits/757581a6f48249b488aeed8cbb61cec9e8695f66 Added a test for the condition too.